packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -459,12 +352,261 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         });
     }
 
-    private async fetchSnapshot(snapshotOptions: string, tokenFetchOptions: TokenFetchOptions) {
+    private isTreesLatestSecondCallEvent(tokenFetchOptions: TokenFetchOptions) {
+        if (tokenFetchOptions.refresh) {
+            // This is the most critical code path for boot.
+            // If we get incorrect / expired token first time, that adds up to latency of boot
+            this.logger.sendErrorEvent({ eventName: "TreeLatest_SecondCall", hasClaims: !!tokenFetchOptions.claims });
+        }
+    }
+
+    private async fetchSnapshotWithMultiPartRequest(
+        tokenFetchOptions: TokenFetchOptions,
+        hostPolicy: ISnapshotOptions,
+    ): Promise<IOdspSnapshot> {
+        let cachedSnapshot: IOdspSnapshot | undefined;
+
+        // No need to ask cache twice - if first request was unsuccessful, cache unlikely to have data on second turn.
+        if (tokenFetchOptions.refresh) {
+            cachedSnapshot = await this.fetchSnapshotWithMultiPartRequestCore(hostPolicy, tokenFetchOptions);
```

Comment:
Double checking - this is called only for the GetLatest call I assume; none of the other calls to SPO (CreateNew, PostSummary,e tc.) use this POST API, correct?

Reply:
To be clear, we can try to switch them over as well later if they are all hitting the OPTIONS calls; but we can use GetLatest to prove out that the approach works with a high degree of confidence.

Reply:
Yes, this is only for getLatest call. We can shift other calls to this form according to your plan.